dashboard/client/src/common/CodeDialogButton/CodeDialogButton.tsx

```diff
@@ -67,9 +72,15 @@ export const CodeDialogButton = ({
           }}
         >
           <Card variant="outlined">
-            <Typography className={classes.configText}>
-              {typeof code === "string" ? code : yaml.dump(code, { indent: 2 })}
-            </Typography>
+            {children ? (
+              children
```

Comment:
I believe the original design for [CodeDialogButton.tsx](https://github.com/ray-project/ray/pull/37118/files#diff-7d2253de348ee23a6bab04106030caf027a027e0b0ad536cbfd33088571a5a8a) is not for displaying a arbitrary HTML element. I could make parse the `customResource` as an object and pass in `CodeDialogButton` without change the `CodeDialogButton` component. What do you think? @alanwguo

Reply:
Can you provide an example of what the text will look like? I think that's reasonable.

Reply:
Sure, the text to display for `customResource` is a group of html elements, like 
<p> custom resouce</p>
<div> CPU: .... </div>
<div> Memory ...</div>

Reply:
We used CodeDialogButtonWithPreview in this PR